From: Jim Rollenhagen <jim@jimrollenhagen.com>
Date: Tue, 8 Sep 2015 13:41:23 +0000 (+0000)
Subject: Replace remaining usage of exc.args[0]
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fironic.git;a=commitdiff_plain;h=7e39461cfb539c7a01e1f44dddda1e63a30edfb4

Replace remaining usage of exc.args[0]

We've converted most of these to str(exc) by now, just a few stragglers left.

Closes-Bug: #1493233

Change-Id: I99196d217a8e20f5a298cadb2b807bd40eea46b5
---

diff --git a/ironic/drivers/modules/ipmitool.py b/ironic/drivers/modules/ipmitool.py
index ff58612..ccfe494 100644
--- a/ironic/drivers/modules/ipmitool.py
+++ b/ironic/drivers/modules/ipmitool.py
@@ -401,7 +401,7 @@ def _exec_ipmitool(driver_info, command):
             except processutils.ProcessExecutionError as e:
                 with excutils.save_and_reraise_exception() as ctxt:
                     err_list = [x for x in IPMITOOL_RETRYABLE_FAILURES
-                                if x in e.args[0]]
+                                if x in str(e)]
                     if ((time.time() > end_time) or
                         (num_tries == 0) or
                         not err_list):
diff --git a/ironic/drivers/modules/msftocs/msftocsclient.py b/ironic/drivers/modules/msftocs/msftocsclient.py
index 01f8a7b..9ba7541 100644
--- a/ironic/drivers/modules/msftocs/msftocsclient.py
+++ b/ironic/drivers/modules/msftocs/msftocsclient.py
@@ -71,7 +71,7 @@ class MSFTOCSClientApi(object):
         except requests_exceptions.RequestException as ex:
             LOG.exception(_LE("HTTP call failed: %s"), ex)
             raise exception.MSFTOCSClientApiException(
-                _("HTTP call failed: %s") % ex.args[0])
+                _("HTTP call failed: %s") % str(ex))
 
         xml_response = response.text
         LOG.debug("Call to %(url)s got response: %(xml_response)s",
diff --git a/ironic/drivers/modules/seamicro.py b/ironic/drivers/modules/seamicro.py
index 5f26a86..8aae9ab 100644
--- a/ironic/drivers/modules/seamicro.py
+++ b/ironic/drivers/modules/seamicro.py
@@ -580,8 +580,8 @@ class Management(base.ManagementInterface):
         except seamicro_client_exception.ClientException as ex:
             LOG.error(_LE("Seamicro set boot device failed for node "
                           "%(node)s with the following error: %(error)s"),
-                      {'node': task.node.uuid, 'error': ex.args[0]})
-            raise exception.IronicException(message=ex.args[0])
+                      {'node': task.node.uuid, 'error': str(ex)})
+            raise exception.IronicException(message=str(ex))
 
     def get_boot_device(self, task):
         """Get the current boot device for the task's node.
diff --git a/ironic/tests/drivers/test_ipmitool.py b/ironic/tests/drivers/test_ipmitool.py
index 4a96855..5e00c33 100644
--- a/ironic/tests/drivers/test_ipmitool.py
+++ b/ironic/tests/drivers/test_ipmitool.py
@@ -348,7 +348,7 @@ class IPMIToolPrivateMethodTestCase(db_base.DbTestCase):
             ValueError,
             self._test__make_password_file,
             mock_sleep, 12345, ValueError('we should fail'))
-        self.assertEqual('we should fail', result.args[0])
+        self.assertEqual('we should fail', str(result))
 
     @mock.patch.object(tempfile, 'NamedTemporaryFile',
                        new=mock.MagicMock(side_effect=OSError('Test Error')))
@@ -367,7 +367,7 @@ class IPMIToolPrivateMethodTestCase(db_base.DbTestCase):
         result = self.assertRaises(
             OverflowError,
             self._test__make_password_file, mock_sleep, 12345)
-        self.assertEqual('Test Error', result.args[0])
+        self.assertEqual('Test Error', str(result))
 
     def test__make_password_file_write_exception(self, mock_sleep):
         # Test exception in _make_password_file for write()
